tst/EnergyPlus/unit/UnitarySystem.unit.cc

```diff
@@ -6202,7 +6208,71 @@ Curve:Biquadratic,
+    EXPECT_NEAR(
+        state->dataZoneEnergyDemand->ZoneSysMoistureDemand(ControlZoneNum).RemainingOutputReqToDehumidSP, -0.0006, 0.00001); // kg moisture per sec
+    EXPECT_NEAR(state->dataZoneEnergyDemand->ZoneSysMoistureDemand(ControlZoneNum).RemainingOutputReqToDehumidSP * HgAir, latOut, 55.0); // Watts
+
+    // not sure why the above control is loose, i.e., 11 W on sensible and 55 W on latent.
+    EXPECT_NEAR(sensOut, 1010.6, 0.1);                                 // Watts
+    EXPECT_NEAR(latOut, -1477.7, 0.1);                                 // Watts
+    EXPECT_NEAR(state->dataUnitarySystems->QToHeatSetPt, 1000.0, 0.1); // heating load
+    EXPECT_NEAR(thisSys->m_MoistureLoadPredicted, -1467.1, 0.1);       // dehumidification control type = CoolReheat so MoistureLoad < 0
+
+    // results using hand calcs
+    Real64 CpAir = Psychrometrics::PsyCpAirFnW(state->dataLoopNodes->Node(ControlZoneNum).HumRat);
+    Real64 DeliveredSensibleCapacity = state->dataLoopNodes->Node(thisSys->AirOutNode).MassFlowRate * CpAir *
+                                       (state->dataLoopNodes->Node(thisSys->AirOutNode).Temp - state->dataLoopNodes->Node(ControlZoneNum).Temp);
+    EXPECT_NEAR(DeliveredSensibleCapacity, 1010.6, 0.001);                     // actual delivered capacity
+    EXPECT_NEAR(DeliveredSensibleCapacity, thisSys->m_SensibleLoadMet, 0.001); // Watts - heating
+
+    Real64 RoomDeltaW = state->dataLoopNodes->Node(thisSys->AirOutNode).HumRat - state->dataLoopNodes->Node(ControlZoneNum).HumRat;
+    Real64 OutDeltaW = state->dataLoopNodes->Node(thisSys->CoolCoilOutletNodeNum).HumRat - state->dataLoopNodes->Node(ControlZoneNum).HumRat;
+    Real64 OutMassFlow = state->dataLoopNodes->Node(thisSys->AirOutNode).MassFlowRate;
+    Real64 LatentOutput = OutDeltaW * OutMassFlow * HgAir;
+    EXPECT_NEAR(OutDeltaW, -0.0003193, 0.0000001);
+    EXPECT_NEAR(OutDeltaW, RoomDeltaW, 0.0000001);
+    EXPECT_NEAR(LatentOutput, -1477.1, 0.1);
+
+    Real64 ExcessSensibleCapacity =
+        state->dataLoopNodes->Node(thisSys->AirOutNode).MassFlowRate * CpAir *
+        (state->dataLoopNodes->Node(thisSys->CoolCoilOutletNodeNum).Temp - state->dataLoopNodes->Node(ControlZoneNum).Temp);
+    EXPECT_NEAR(ExcessSensibleCapacity, -17268.1, 0.1);
+    EXPECT_NEAR(state->dataHeatingCoils->HeatingCoil(thisSys->m_HeatingCoilIndex).HeatingCoilRate, 0.0, 0.1);
+    EXPECT_NEAR(state->dataHeatingCoils->HeatingCoil(thisSys->m_SuppHeatCoilIndex).HeatingCoilRate, 18268.1, 0.1);
+    Real64 ReheatNeded = sensOut - ExcessSensibleCapacity;
+    EXPECT_NEAR(ReheatNeded, 18268.1, 11.0); // same 11 watt difference as above, is this error due to tolerance?
+
+    // remove heating coil to test cooling only application
+    thisSys->m_HeatCoilExists = false;
+    thisSys->m_HeatingCoilIndex = 0;
+    thisSys->HeatCoilInletNodeNum = 0;
+    thisSys->HeatCoilOutletNodeNum = 0;
```

Comment:
Beefed up this unit test a little to better understand what was happening and added this last test to remove heating coil and preform calculations again. This unit test fails without this branches code change.